models/articles-model.js

```diff
@@ -31,4 +31,25 @@ exports.selectAllArticles = () => {
       const articles = result.rows;
       return articles;
     });
-};
\ No newline at end of file
+};
+
+exports.updateArticleVotes = (articleId, votes) => {
+  if (typeof articleId !== "number" || typeof votes !== "number") {
+    return Promise.reject({ status: 400, msg: "Bad request" });
+  }
+  return exports
+    .selectArticles(articleId)
+    .then(() => {
+      return db.query(
+        `UPDATE articles SET votes = votes+$1 WHERE article_id = $2 RETURNING *`,
+        [votes, articleId]
+      );
+    })
+    .then((result) => {
+      const article = result.rows[0];
+      if (!article) {
+        return Promise.reject({ status: 404, msg: "Page not found" });
```

Comment:
If you're using select articles to make sure the article exists, this reject will never fire as the catch in select articles will fire if there is no rows.